README.md

```diff
@@ -1,50 +1,96 @@
+{
+  "id": "A string id of the merkle tree, can be random",
+  "metadata": {
+  "info": "a key value mapping allowing you to add information"
+  },
+  "root": "The merkle root of the tree",
+  "tree": [
+       {
+              "account": "The address of the claimer",
+              "reward": "The address of the reward token",
+              "claimable": "The claimable amount as a big number",
+              "proof": ["0x1...", "0x2...", "...", "0xN..."]
+      }
+  ]
+}
 ```
 
-Install it:
-```bash
-foundryup
-```
+- We recommend not including spaces or new lines when uploading to IPFS, to ensure a consistent method of uploading your JSON file.
 
-Install dependencies:
-```bash
-git submodule update --init --recursive
-```
+## Owner Specifications
 
-Now you can run tests, using forge:
-```bash
-forge test
-```
+- The URD is an owner-managed contract, meaning the owner has full control over its distribution. Specifically, the owner can bypass all time-locked functions, modify the time-lock, add or remove updaters, and revoke the pending value at their discretion.
+- If the owner is set to a zero address, the contract becomes ownerless. In this scenario, only updaters can submit a root. Furthermore, the time-lock value becomes unchangeable, and pending values cannot be removed.
+- If there are neither owner nor updaters, the URD becomes immutable. This is useful for creating a one-time distribution that must remain unchanged. This can be accomplished at the contract's creation by providing only a Merkle root and an optional IPFS hash. After this, the contract's sole functionality is to claim rewards.
+- It is possible to create a URD with no root, owner, or updaters. While this might seem pointless, it is the URD creator's responsibility to configure the URD correctly.
+
+## Updaters Specifications
+
+- Multiple updaters can be defined for a single URD. The owner can add or remove updaters at any time, instantly, without a timelock.
+- All updaters share a single pending value. This means they can override the pending value (if any) at any time.
+- If a pending value is not used as the main root, any submissions by updaters will override the pending value.
+
+Having multiple updaters can lead to situations where the pending values are subject to multiple concurrent propositions. The owner must manage this scenario to maintain the URD's functionality. We recommend not having different reward distribution strategies in a single URD. All updaters should agree on a distribution mechanism and a root.The use case for having multiple updaters is to enhance resilience and security.
+
+## Considerations for Merkle Tree
+
+- The claimable amount for a given user must always exceed the amount provided in the previous Merkle tree. If a claimer has claimed an amount higher than the amount in the Merkle tree (if claimed from a previous root), the claim will revert with a "root misconfigured" error.
+- TODO: Define the list of invariants for a new root.
```

Comment:
todo

Reply:
Yes I know, imo it is not in the scope of this PR. I have to think more in-depth about all the checks we can do on a specific Merkle tree.